File: 1_two_sum/two_sum.ts
function twoSum(nums: number[], target: number): number[] {
    // Target = A + B
    // B = Target - A
    const hashMap: Map<number,number> = new Map();
    for (let i = 0; i< nums.length; i++){
        let A: number = nums[i];
        let B: number = target - A;
        if (hashMap.has(B)){
            return [hashMap.get(B),i];
        }
        hashMap.set(A,i);
    }
    return null;
};

File: 1_two_sum/two_sum.go
func twoSum(nums []int, target int) []int {
    hashmap := map[int]int{}
    // Target = A + B
    // Target - A = B
    for index, A := range(nums){
        B := target - A
        _, present := hashmap[B]
        if present{
            return []int{hashmap[B],index}
        }
        hashmap[A] = index
    }
    return nil
}
